src/types.ts

```diff
@@ -1051,22 +1051,14 @@ export class Signature {
   isAssignableTo(target: Signature, checkCompatibleOverride: bool = false): bool {
     let thisThisType = this.thisType;
     let targetThisType = target.thisType;
-    if (checkCompatibleOverride) {
-      // check kind of `this` type
-      if (thisThisType) {
-        if (!targetThisType || !thisThisType.canExtendOrImplement(targetThisType)) {
-          return false;
-        }
-      } else if (targetThisType) {
-        return false;
-      }
-    } else {
-      // check `this` type (invariant)
-      if (thisThisType) {
-        if (targetThisType != targetThisType) return false;
-      } else if (targetThisType) {
-        return false;
-      }
+
+    if (thisThisType != null && targetThisType != null){
```

Comment:
You are very close. You're missing a space before the opening brace on this line, and you're missing two spaces in the `else if` line: one right before `else if`, and one before the opening brace.

I should have noticed this earlier, but use `thisThisType && targetThisType` instead of `thisThisType != null && targetThisType != null`.

@dcodeIO Is this ternary I proposed okay with you?

Reply:
Done

Reply:
@dcodeIO @MaxGraey ping maintainer.